### ISSUE TITLE
Grafana Loki Integration: Health check fails when Loki is up and functional

### ISSUE THREAD

**Issue by MegaShinySnivy** [grafana/grafana#79846]
### What happened?

When attempting to add a new loki datasource via grafana, I get this error in the console, and a 'failed to connect' on the webUI.

`logger=tsdb.loki endpoint=checkHealth pluginId=loki dsName=Loki dsUID=P8E80F9AEF21F6940 uname=admin fromAlert=false t=2023-12-21T09:52:28.788951687Z level=error msg="Error received from Loki" duration=3.855835ms stage=databaseRequest statusCode=400 contentLength=65 start=1970-01-01T00:00:01Z end=1970-01-01T00:00:04Z step=1s query=vector(1)+vector(1) queryType=instant direction=backward maxLines=0 supportingQueryType=none lokiHost=loki-stack:3100 lokiPath=/loki/api/v1/query status=error error="parse error at line 1, col 1: syntax error: unexpected IDENTIFIER"`

This happens even though my Loki instance is very much up and healthy, and responds to other LogQL queries for actual data. 

### What did you expect to happen?

The query returns successfully, and the datasource is added.

### Did this work before?

It did, however I am unsure which version worked as this happened after I migrated from one kubernetes cluster to another.

### How do we reproduce it?

1. Create a generic K3s cluster with some sort of storage provider and CNI
2. Install grafana on it using the values found [here](https://github.com/MegaShinySnivy/Scaling-Snakes/blob/main/kubernetes/apps/monitoring/grafana/app/helmrelease.yaml), loki via the values found [here](https://github.com/MegaShinySnivy/Scaling-Snakes/blob/main/kubernetes/apps/monitoring/kube-loki-stack/app/helmrelease.yaml), and a testing application of your choice
3. Test the added Loki data source, and find that the query fails

### Is the bug inside a dashboard panel?

No

### Environment (with versions)?

Grafana: 10.2.2
OS: N/A, running via K3s version v1.28.4+k3s2, with Cillium CNI
Browser: Firefox 121.0

### Grafana platform?

Kubernetes

### Datasource(s)?

Loki

**Comment by MegaShinySnivy** [comment]
Found my issue. The version of loki bundled with loki-stack is extremely out of date. I'll make a bug report over there that the version needs to be bumped

**Comment by mrbnd** [comment]
Hello @MegaShinySnivy. I met the same problem, according to this info [community.grafana](https://community.grafana.com/t/difference-between-helm-charts-loki-and-loki-stack/87380/2) "loki-stack is no longer maintained". Config from this [page](https://grafana.com/docs/loki/latest/setup/install/helm/install-monolithic/) + promtail default config `helm upgrade -install promtail -n $NAMESPACE grafana/promtail` works for me.

**Comment by MegaShinySnivy** [comment]
I see that it is community-maintained on the helm repository itself, is this not actually the case? If not, then perhaps a warning on that helm chart should be considered.

**Comment by gawbul** [comment]
See my comment [here](https://github.com/grafana/loki/issues/11557#issuecomment-2012879496) on how I resolved this.

**Comment by JanBenisek** [comment]
I got the same issue when trying to add Loki data source to Grafana.

> logger=tsdb.loki endpoint=CheckHealth t=2024-11-06T17:43:29.739592236Z level=error msg="Loki health check failed" error="error from loki: parse error at line 1, col 1: syntax error: unexpected IDENTIFIER"

Adding the tag resolved it!
```yaml
loki:
  image:
    tag: '2.9.3'
```
FYI I am on `grafana/loki-stack` chart version `2.10.2` and Grafana from `kube-prometheus-stack` with `65.8.1`.
Thanks y'all for contributing!

**Comment by rajpratik71** [comment]
> I got the same issue when trying to add Loki data source to Grafana.
> 
> > logger=tsdb.loki endpoint=checkHealth pluginId=loki dsName=loki dsUID=ce2yl5djlou80e uname=admin fromAlert=false t=2024-11-06T17:43:29.739484443Z level=error msg="Error received from Loki" duration=3.105666ms stage=databaseRequest statusCode=400 contentLength=65 start=1970-01-01T00:00:01Z end=1970-01-01T00:00:04Z step=1s query=vector(1)+vector(1) queryType=instant direction=backward maxLines=0 supportingQueryType=none lokiHost=loki-stack:3100 lokiPath=/loki/api/v1/query status=error error="parse error at line 1, col 1: syntax error: unexpected IDENTIFIER"
> 
> Adding the tag resolved it!
> 
> loki:
>   image:
>     tag: '2.9.3'
> FYI I am on `grafana/loki-stack` chart version `2.10.2` and Grafana from `kube-prometheus-stack` with `65.8.1`. Thanks y'all for contributing!

Below tag also works which have dependencies update and CVE fixes.

```
loki:
  image:
    tag: '2.9.15'
```

**Comment by sonaeccit** [comment]
Same issue resolved by explicitly mentioning image version 2.9.3.
